### ISSUE TITLE
question on  conversion from sigmoid logit to softmax

### ISSUE THREAD

**Issue by hengck23** [fhopfmueller/bi-tempered-loss-pytorch#1]
for a binary classification, given a logit x value, we compute the sigmoid value as  sigmoid(x). This is the same as using softmax ( [x,-x] ). 

However, you seem to be using softmax ( [x,0] ) instead. Why is this so?

```
def tempered_sigmoid(activations, t, num_iters = 5):
        internal_activations = torch.stack([activations, torch.zeros_like(activations)], dim=-1)
       
       #shoudn't it be this ????
        internal_activations = torch.stack([activations,- activations], dim=-1)
```

**Comment by fhopfmueller** [comment]
Thanks for the comment!
I believe we have sigmoid(x) = softmax([x, 0]), so the code should be correct. Test with
```
import torch
a = torch.randn(10)
sigmoid_a = a.sigmoid()
softmax_a_0 = torch.softmax( 
    torch.stack( [a, torch.zeros_like(a)], dim=-1),
    dim=-1)[:, 0]
assert (sigmoid_a - softmax_a_0).abs().max() < 1e-5
```

**Comment by hengck23** [comment]
thank you for your answer. i figure it out and think both will work:

```
sigmoid (x) = 1/( 1 + exp(-x)) = exp(0) /(exp(0)+ exp(-x)) = softmax([0,x])[0]

or 

sigmoid (x) = 1/( 1 + exp(-x)) = exp(x/2)/( exp(x/2) + exp(-x/2)) = softmax([x/2,-x/2])[0]

```
